File: docs/legacy(old PACE)/C#/PaceAPISample/sample/CreateEstimatePart.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using efipaceservices;
using System.Web.Services.Protocols;

namespace Pace_Web_Service_SDK.sample
{
    public class CreateEstimatePart
    {
        public static void Run()
        {
            InvokeActionPortType invokeActionHttpBinding = PaceClient.getInvokeActionHttpBinding();
            Console.WriteLine("Enter the estimate id for which the estimate part has to be added");
            String estimateNumber = Console.ReadLine();

            EstimatePartInfo estimatePartInfo = createEstimatePartInfo(estimateNumber);
            try
            {
                EstimatePart estimatePart = invokeActionHttpBinding.addEstimatePart(estimatePartInfo);
                Console.WriteLine("Estimate Part was successfully added to the estimate Number " + estimatePart.estimate);
            }
            catch (SoapException e)
            {
                Console.WriteLine(e.Message);
            }
        }

        public static EstimatePartInfo createEstimatePartInfo(String estimateNumber)
        {
            EstimatePartInfo estimatePartInfo = new EstimatePartInfo();
            estimatePartInfo.estimateID = estimateNumber;
            estimatePartInfo.foldPattern = "2:1";
            estimatePartInfo.finalSizeH = 12;
            estimatePartInfo.finalSizeW = 22;
            estimatePartInfo.eachOf = 22;
            estimatePartInfo.numPlies = 32;
            estimatePartInfo.grainSpecifications = 3;

            estimatePartInfo.bindingMethod = 1;
            estimatePartInfo.prepressWorkflow = 4;

            estimatePartInfo.product = "FL";

            // estimatePartInfo.setCompositeProduct(5001);
            estimatePartInfo.colorsSide1 = 4;
            estimatePartInfo.colorsSide2 = 4;
            estimatePartInfo.totalColors = 4;
            estimatePartInfo.inkCoverageFront = 3;

            estimatePartInfo.inkCoverageBack = 3;

            estimatePartInfo.quantity1 = 1000;
            estimatePartInfo.quantity2 = 2000;
            estimatePartInfo.quantity1Desc = " qty1 desc ";
            estimatePartInfo.quantity2Desc = " qty2 desc ";

            EstimatePaperInfo estimatePaperInfo = createEstimatePaperInfo();
            EstimatePressInfo estimatePressInfo = createEstimatePressInfo();

            estimatePartInfo.estimatePaperInfo = estimatePaperInfo;
            estimatePartInfo.estimatePressInfo = estimatePressInfo;

            return estimatePartInfo;
        }

        private static EstimatePaperInfo createEstimatePaperInfo()
        {
            EstimatePaperInfo estimatePaperInfo = new EstimatePaperInfo();
            estimatePaperInfo.materialType = "InventoryItem";
            estimatePaperInfo.inventoryItem = SDK.readStringInputFromTheUser("Please enter Inventory Item id for the Estimate : ");
            estimatePaperInfo.weight = 5002;
            estimatePaperInfo.uom = "EA";
            estimatePaperInfo.buySizeH = 25;
            estimatePaperInfo.buySizeW = 30;
            estimatePaperInfo.buySizeGrainDirection = 2;         

            return estimatePaperInfo;
        }

        private static EstimatePressInfo createEstimatePressInfo()
        {
            EstimatePressInfo estimatePressInfo = new EstimatePressInfo();
            estimatePressInfo.primaryPress = 1;
            estimatePressInfo.runMethod = 1;
            estimatePressInfo.runSizeH = 25;
            estimatePressInfo.runSizeW = 30;
            estimatePressInfo.runSizeGrainDirection = 2;
            return estimatePressInfo;
        }
    }
}

File: docs/legacy(old PACE)/unsupported/VB.NET/PaceWebServiceSDK/sample/CreateEstimatePart.vb
Imports PaceWebServiceSDK.efipaceservices

Public Class CreateEstimatePart

    Public Shared Sub Run()
        Dim invokeActionHttpBinding As InvokeActionHttpBinding = PaceClient.getInvokeActionHttpBinding()
        Console.WriteLine("Enter the estimate id for which the estimate part has to be added")
        Dim estimateNumber As String = Console.ReadLine()

        Dim estimatePartInfo As EstimatePartInfo = createEstimatePartInfo(estimateNumber)

        Try
            Dim estimatePart As EstimatePart = invokeActionHttpBinding.addEstimatePart(estimatePartInfo)
            Console.WriteLine("Estimate Part was successfully added to the estimate " + estimatePart.estimate.ToString)
        Catch ex As Exception
            Console.WriteLine(ex.Message)
        End Try

    End Sub

    Public Shared Function createEstimatePartInfo(ByVal estimateNumber As String) As EstimatePartInfo
        Dim estimatePartInfo As EstimatePartInfo = New EstimatePartInfo()
        estimatePartInfo.estimateID = estimateNumber
        estimatePartInfo.foldPattern = "2:1"
        estimatePartInfo.finalSizeH = 12
        estimatePartInfo.finalSizeW = 22
        estimatePartInfo.eachOf = 22
        estimatePartInfo.numPlies = 32
        estimatePartInfo.grainSpecifications = 3

        estimatePartInfo.bindingMethod = 1
        estimatePartInfo.prepressWorkflow = 4

        estimatePartInfo.product = "FL"

        ' estimatePartInfo.setCompositeProduct(5001);
        estimatePartInfo.colorsSide1 = 4
        estimatePartInfo.colorsSide2 = 4
        estimatePartInfo.totalColors = 4
        estimatePartInfo.inkCoverageFront = 3

        estimatePartInfo.inkCoverageBack = 3

        estimatePartInfo.quantity1 = 1000
        estimatePartInfo.quantity2 = 2000
        estimatePartInfo.quantity1Desc = " qty1 desc "
        estimatePartInfo.quantity2Desc = " qty2 desc "

        Dim estimatePaperInfo As EstimatePaperInfo = createEstimatePaperInfo()
        Dim estimatePressInfo As EstimatePressInfo = createEstimatePressInfo()

        estimatePartInfo.estimatePaperInfo = estimatePaperInfo
        estimatePartInfo.estimatePressInfo = estimatePressInfo

        Return estimatePartInfo
    End Function

    Private Shared Function createEstimatePaperInfo() As EstimatePaperInfo
        Dim estimatePaperInfo As EstimatePaperInfo = New EstimatePaperInfo()
        estimatePaperInfo.materialType = "InventoryItem"
        estimatePaperInfo.inventoryItem = "1001"
        estimatePaperInfo.weight = 5002
        estimatePaperInfo.uom = "EA"

        Return estimatePaperInfo
    End Function

    Private Shared Function createEstimatePressInfo() As EstimatePressInfo
        Dim estimatePressInfo As EstimatePressInfo = New EstimatePressInfo()
        estimatePressInfo.primaryPress = 1
        estimatePressInfo.runMethod = 1
        Return estimatePressInfo
    End Function

End Class
